src/System.Management.Automation/engine/hostifaces/ConnectionBase.cs

```diff
@@ -37,22 +37,14 @@ internal abstract class RunspaceBase : Runspace
         /// <exception cref="System.ArgumentNullException">
         /// host is null.
         /// </exception>
-        /// <param name="runspaceConfiguration">
-        /// configuration information for this minshell.
-        /// </param>
-        protected RunspaceBase(PSHost host, RunspaceConfiguration runspaceConfiguration)
+        protected RunspaceBase(PSHost host)
         {
             if (host == null)
             {
                 throw PSTraceSource.NewArgumentNullException("host");
             }
-            if (runspaceConfiguration == null)
-            {
-                throw PSTraceSource.NewArgumentNullException("runspaceConfiguration");
-            }
-
+            InitialSessionState = InitialSessionState.CreateDefault2();
```

Comment:
Yes, this is a breaking change.  Our test coverage may not be sufficient to cover this so I need someone with experience here to point out how this may have adverse effects.

Reply:
If we want to mimic the existing behavior, then we should use `InitialSessionState.CreateDefault()` here. When using `RunspaceConfiguration`. all built-in powershell modules are loaded as snapins, so you will get nothing when running `Get-Module` even after executing `Get-Process`. With `CreateDefault()`, built-in powershell modules are loaded as snapins as well. But with `CreateDefault2()`, only the `Microsoft.PowerShell.Core` is loaded as snapin and the rest are loaded as modules, like a regular powershell console session does.